xdsl/dialects/wasm/attrs.py

```diff
@@ -0,0 +1,349 @@
+        super().__init__((ArrayAttr(args), ArrayAttr(res)))
+
+    @classmethod
+    def parse_parameters(cls, parser: AttrParser) -> Sequence[Attribute]:
+        with parser.in_angle_brackets():
+            args = parser.parse_comma_separated_list(
+                AttrParser.Delimiter.PAREN, lambda: parser.parse_attribute()
+            )
+            parser.parse_punctuation("->")
+            res = parser.parse_comma_separated_list(
+                AttrParser.Delimiter.PAREN, lambda: parser.parse_attribute()
+            )
+            return (ArrayAttr(args), ArrayAttr(res))
+
+    def print_parameters(self, printer: Printer) -> None:
+        with printer.in_angle_brackets():
+            printer.print("(")
+            printer.print_list(self.args.data, lambda x: printer.print_attribute(x))
+            printer.print(") -> (")
+            printer.print_list(self.res.data, lambda x: printer.print_attribute(x))
+            printer.print(")")
+
+
+##==------------------------------------------------------------------------==##
+# WebAssembly tables and memories
+##==------------------------------------------------------------------------==##
+
+
+@irdl_attr_definition
+class WasmLimits(ParametrizedAttribute):
+    """
+    wasm> Limits classify the size range of resizable storage associated with
+    memory types and table types. If no maximum is given, the respective storage
+    can grow to any size.
+    """
+
+    name = "wasm.limits"
+
+    min: ParameterDef[I32Attr]
+    max: ParameterDef[I32Attr | UnitAttr]
+
+    def __init__(self, min: I32Attr | int, max: I32Attr | int | None):
+        min_attr = min if isinstance(min, IntegerAttr) else IntegerAttr(min, 32)
+        max_attr = (
+            max
+            if isinstance(max, IntegerAttr)
+            else IntegerAttr(max, 32) if max is not None else UnitAttr()
+        )
+        super().__init__((min_attr, max_attr))
+
+
+@irdl_attr_definition
+class WasmMemoryType(ParametrizedAttribute):
+    """
+    wasm> Memory types classify linear memories and their size range. The limits
+    constrain the minimum and optionally the maximum size of a memory. The limits
+    are given in units of page size.
+    """
+
+    name = "wasm.mem"
+
+    limits: ParameterDef[WasmLimits]
+
+    def __init__(self, limits: WasmLimits):
+        super().__init__((limits,))
+
+
+@irdl_attr_definition
+class WasmTableType(ParametrizedAttribute):
+    """
+    wasm> Table types classify tables over elements of reference type within a
+    size range. Like memories, tables are constrained by limits for their
+    minimum and optionally maximum size. The limits are given in numbers of
+    entries.
+    """
+
+    name = "wasm.table"
+
+    elements: ParameterDef[WasmRefType]
+    limits: ParameterDef[WasmLimits]
+
+    def __init__(self, elem: WasmRefType, limits: WasmLimits):
+        super().__init__((elem, limits))
+
+
+class WasmMutEnum(StrEnum):
+    Const = auto()
+    Var = auto()
```

Comment:
```suggestion
    CONST = auto()
    VAR = auto()
```

Reply:
Do you mean that this should be caps because they are constants?

Reply:
Yep, this is the Python convention https://docs.python.org/3/library/enum.html